homeassistant/components/media_player/cambridgeaudio.py

```diff
@@ -0,0 +1,382 @@
+        return self._audio_source
+
+    @property
+    def source_list(self):
+        """Return a list of available Presets."""
+        return self._source_list
+
+    @property
+    def supported_features(self):
+        """Return supported features."""
+        features = SUPPORT_CAMBRIDGE_COMMON
+        if self._audio_source == "media player":
+            features = features | SUPPORT_CAMBRIDGE_FILEMEDIA
+        if self._support_volumecontrol:
+            features = features | SUPPORT_CAMBRIDGE_PREAMP
+        return features
+
+    @property
+    def media_content_type(self):
+        """Return the media content type."""
+        return MEDIA_TYPE_MUSIC
+
+    @property
+    def media_image_url(self):
+        """Return image URI of current playing media."""
+        return self._album_art
+
+    @property
+    def media_artist(self):
+        """Return artist of current playing media."""
+        return self._artist
+
+    @property
+    def media_album_name(self):
+        """Return album of current playing media."""
+        return self._album
+
+    @property
+    def media_track(self):
+        """Return track number of current playing media."""
+        return self._trackno
+
+    @property
+    def media_title(self):
+        """Return title of current playing media."""
+        return self._title
+
+    @property
+    def media_position(self):
+        """Return position of current playing media in seconds."""
+        if self._state is not STATE_PLAYING:
+            return None
+        return self._position
+
+    @property
+    def media_duration(self):
+        """Return the duration of current playing media in seconds."""
+        if self._state is not STATE_PLAYING:
+            return None
+        return self._duration
+
+    @property
+    def media_position_updated_at(self):
+        """When was the position of the current playing media valid.
+
+        Returns value from homeassistant.util.dt.utcnow().
+        """
+        if self._state in [STATE_PLAYING, STATE_PAUSED]:
+            return self._position_updated_at
+
+    @property
+    def volume_level(self):
+        """Current volume level (0..1)."""
+        return float(self._volume) / float(self._volume_max)
+
+    def clear_playlist(self):
+        """Clear the playlist."""
+        pass
+
+    def media_next_track(self):
+        """Skip to the next track, when in media player mode."""
+        if self._audio_source == "media player":
+            self._smdevice.trnsprt_next()
+
+    def media_previous_track(self):
+        """Skip to the previous track, when in media player mode."""
+        if self._audio_source == "media player":
+            self._smdevice.trnsprt_prev()
+
+    def media_pause(self):
+        """Pause playing the current media."""
+        # there is no "pause" with internet radio streams
+        if self._audio_source == "media player":
+            self._smdevice.trnsprt_pause()
+            self._state = STATE_PAUSED
```

Comment:
State will be updated directly after the service call by the `update` method. We can probably remove this here and in all the other actuator methods.